components/match2/wikis/arenafps/match_group_input_custom.lua

```diff
@@ -39,349 +31,171 @@ local CustomMatchGroupInput = {}
+function MatchFunctions.calculateMatchScore(maps, bestOf)
+	return function(opponentIndex)
+		if bestOf == 1 then
+			if not maps[1] or not maps[1].scores then
+				return
 			end
+			return maps[1].scores[opponentIndex]
 		end
+		return MatchGroupInputUtil.computeMatchScoreFromMapWinners(maps, opponentIndex)
 	end
-
-	-- see if match should actually be finished if score is set
-	if isScoreSet and not Logic.readBool(match.finished) and match.timestamp ~= DateExt.defaultTimestamp then
-		local threshold = match.dateexact and 30800 or 86400
-		if match.timestamp + threshold < NOW then
-			match.finished = true
-		end
-	end
-
-	-- apply placements and winner if finshed
-	if not Logic.isEmpty(match.winner) or Logic.readBool(match.finished) then
-		match, opponents = CustomMatchGroupInput.getResultTypeAndWinner(match, opponents)
-	end
-
-	-- Update all opponents with new values
-	for opponentIndex, opponent in pairs(opponents) do
-		match['opponent' .. opponentIndex] = opponent
-	end
-	return match
 end
 
--- Get Playerdata for non-team opponents
----@param player table
----@return boolean
-function CustomMatchGroupInput._playerIsBye(player)
-	return (player.name or ''):lower() == MATCH_BYE or (player.displayname or ''):lower() == MATCH_BYE
+---@param match table
+---@return table
+function MatchFunctions.getLinks(match)
+	return {
+		preview = match.preview,
+		quakehistory = match.quakehistory and ('http://www.quakehistory.com/en/matches/' .. match.quakehistory) or nil,
+		dbstats = match.dbstats and ('https://quakelife.ru/diabotical/stats/matches/?matches=' .. match.dbstats) or nil,
+		qrindr = match.qrindr and ('https://qrindr.com/match/' .. match.qrindr) or nil,
+		esl = match.esl and ('https://play.eslgaming.com/match/' .. match.esl) or nil,
+		stats = match.stats,
+	}
 end
 
 --
 -- map related functions
 --
 
--- Parse extradata information
 ---@param map table
+---@param opponentCount integer
 ---@return table
-function mapFunctions.getExtraData(map)
-	map.extradata = {
+function MapFunctions.getExtraData(map, opponentCount)
+	return {
 		comment = map.comment,
 	}
-	return map
 end
 
--- Calculate Score and Winner of the map
----@param map table
----@return table
-function mapFunctions.getScoresAndWinner(map)
-	map.scores = {}
-	local indexedScores = {}
-	for scoreIndex = 1, MAX_NUM_OPPONENTS do
-		-- read scores
-		local score = map['score' .. scoreIndex] or map['t' .. scoreIndex .. 'score']
-		local obj = {}
-		if not Logic.isEmpty(score) then
-			if Logic.isNumeric(score) then
-				obj.status = 'S'
-				obj.score = score
-			elseif Table.includes(ALLOWED_STATUSES, score) then
-				obj.status = score
-				obj.score = -1
-			end
-			table.insert(map.scores, score)
-			indexedScores[scoreIndex] = obj
-		else
-			break
+---@param winnerInput string|integer|nil
+---@param finished boolean
+---@return fun(opponentIndex: integer): integer?
+function MapFunctions.calculateMapScore(winnerInput, finished)
```

Comment:
This doesn't seem to match the old mapScore func

Reply:
how so?

Reply:
arenafps basically never has 1-0 as a map score, so having a score of 1-0 would be incorrect (correct would be in that case W-FF)

<img width="331" alt="image" src="https://github.com/user-attachments/assets/6e70750b-5ae6-4a24-b0cc-ad0e0a044f46">

Reply:
okay, kicking it in that case